### PR TITLE
Introduce fixRemoveNFTokenAutoTrustLine amendment

### PR DESCRIPTION
**Pull request by scottschurr**
## High Level Overview of Change

It turns out that the feature enabled by the `tfTrustLine` flag on an `NFTokenMint` transaction could be used as a means to attack the NFToken issuer.  Details are in https://github.com/XRPLF/rippled/issues/4300

The `fixRemoveNFTokenAutoTrustLine` amendment in this pull request removes the ability to set the `tfTrustLine` flag on an `NFTokenMint` transaction.

Closes https://github.com/XRPLF/rippled/issues/4300

### Context of Change

There are two commits in this pull request.  They are both needed, but they should be kept separate (not squashed).

- The first commit, by @HowardHinnant, is required to allow the code to be compiled without warnings by the latest version of Xcode (Version 13.4.1.  This commit has already been accepted into the 1.10.0-b1 develop branch.  This first commit does not require review.

- The second commit disables the `tfTrustLine` flag on an NFTokenMint transaction.  This commit needs review.

### Type of Change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Amendment (amendment going live will cause amendment blocking of out-of-date servers)
- [x] Tests (tests were added for the new amendment)
- [x] Requires documentation updates

### CODE REVIEW

File: src/ripple/app/tx/impl/NFTokenMint.cpp [hunk 1]
```diff
@@ -40,7 +40,23 @@ NFTokenMint::preflight(PreflightContext const& ctx)
...
+    // building up any number of TrustLines on the issuer, increasing the
+    // issuer's reserve without bound.
+    //
+    // The fixRemoveNFTokenAutoTrustLine amendment disables minting with the
+    // tfTrustLine flag as a way to prevent the attack.  But until the
+    // amendment passes we still need to keep the old behavior available.
+    std::uint32_t const NFTokeMintMask =
+        ctx.rules.enabled(fixRemoveNFTokenAutoTrustLine) ? tfNFTokenMintMask
+                                                         : tfNFTokenMintOldMask;
+    if (ctx.tx.getFlags() & NFTokeMintMask)
```
Comment (ximinez) [inline src/ripple/app/tx/impl/NFTokenMint.cpp:59]: Nit: Should this variable be spelled `NFTokenMintMask`?

**Review by ximinez — CHANGES_REQUESTED** [review]

File: src/ripple/app/tx/impl/NFTokenMint.cpp [hunk 2]
```diff
@@ -40,7 +40,23 @@ NFTokenMint::preflight(PreflightContext const& ctx)
...
+    //
+    // That capability could be used to attack the NFToken issuer.  It
+    // would be possible for two accounts to trade the NFToken back and forth
+    // building up any number of TrustLines on the issuer, increasing the
+    // issuer's reserve without bound.
+    //
+    // The fixRemoveNFTokenAutoTrustLine amendment disables minting with the
+    // tfTrustLine flag as a way to prevent the attack.  But until the
+    // amendment passes we still need to keep the old behavior available.
+    std::uint32_t const NFTokeMintMask =
```
Comment (nbougalis) [inline src/ripple/app/tx/impl/NFTokenMint.cpp:56]: Minor typo here `NFTokeMintMask` is missing an `n`.

File: src/ripple/app/tx/impl/NFTokenMint.cpp (hunk 1 above)
Comment (scottschurr) [reply src/ripple/app/tx/impl/NFTokenMint.cpp:59]: 😀 🚬 Good catch.  Yes, that's quite the misspelling.

**Review by nbougalis — APPROVED** [review]
Looks good, just a small typo.

File: src/ripple/app/tx/impl/NFTokenMint.cpp (hunk 2 above)
Comment (gadget78) [reply src/ripple/app/tx/impl/NFTokenMint.cpp:56]: its also missing on line 59 too ...

**Comment by scottschurr** [comment]
Typo fixed.  Thanks for catching that.

**Comment by intelliot** [comment]
[Released in version 1.9.4](https://xrpl.org/blog/2022/rippled-1.9.4.html)

**Pull request closed without merging**